src/twisted/names/dns.py

```diff
@@ -1902,6 +1917,74 @@ def __hash__(self):
 
 
 
+@implementer(IEncodable, IRecord)
+class Record_SSHFP(tputil.FancyEqMixin, tputil.FancyStrMixin):
+    """
+    A record containing the fingerprint of an SSH key.
+
+    @type algorithm: L{int}
+    @ivar algorithm: The SSH key's algorithm, such as L{ALGORITHM_RSA}.
+        Note that the numbering used for SSH key algorithms is specific
+        to the SSHFP record, and is not the same as the numbering
+        used for KEY or SIG records.
+
+    @type fingerprintType: L{int}
+    @ivar fingerprintType: The fingerprint type,
+        such as L{FINGERPRINT_TYPE_SHA256}.
+
+    @type fingerprint: L{bytes}
+    @ivar fingerprint: The key's fingerprint, e.g. a 32-byte SHA-256 digest.
+
+    @cvar ALGORITHM_RSA: The algorithm value for C{ssh-rsa} keys.
+    @cvar ALGORITHM_DSS: The algorithm value for C{ssh-dss} keys.
+    @cvar ALGORITHM_ECDSA: The algorithm value for C{ecdsa-sha2-*} keys.
+    @cvar ALGORITHM_Ed25519: The algorithm value for C{ed25519} keys.
+
+    @cvar FINGERPRINT_TYPE_SHA1: The type for SHA-1 fingerprints.
+    @cvar FINGERPRINT_TYPE_SHA256: The type for SHA-256 fingerprints.
+
+    @see: U{RFC 4255 <https://tools.ietf.org/html/rfc4255>}
+          and
+          U{RFC 6594 <https://tools.ietf.org/html/rfc6594>}
+    """
+    fancybasename = "SSHFP"
+    compareAttributes = ('algorithm', 'fingerprintType', 'fingerprint', 'ttl')
+    showAttributes = ('algorithm', 'fingerprintType', 'fingerprint')
+
+    TYPE = SSHFP
+
+    ALGORITHM_RSA = 1
+    ALGORITHM_DSS = 2
+    ALGORITHM_ECDSA = 3
+    ALGORITHM_Ed25519 = 4
+
+    FINGERPRINT_TYPE_SHA1 = 1
+    FINGERPRINT_TYPE_SHA256 = 2
+
+    def __init__(self, algorithm=0, fingerprintType=0, fingerprint=b'', ttl=0):
```

Comment:
I don't like the default values.
What is the reason of doing this?

I think that by default, we should initialize a valid object and document the default values ... or else make the argument mandatory.

Reply:
I would agree, but the way that t.n.dns is written, each record type has to be instantiable with no arguments except for `ttl`. (See the `parseRecords` method of `Message`.)

There are certainly cleaner ways to write that, and I'd even be interested in implementing one if you think the Twisted community would be willing to accept a bunch of compatibility-breaking changes for such a thing, but I think that's something that should be done in a different PR.

Reply:
Just asking.  Do we need to create by default an instance which is not valid?

This PR might need a review from someone who cares about DNS :)

I have never used the DNS part of Twisted, so in these matters of API design maybe is best to have someone else approve or reject it.

Reply:
I think most of the arguments *should* be mandatory, because there isn't really a good default value. I think we agree on that.

Because it's not possible to make them mandatory (without widespread compatibility-breaking changes to the module) I think that giving them invalid default values is the second-best option. It would be possible to give them arbitrary, but valid, defaults, but I wouldn't want anyone to write code that relies on those defaults. If `fingerprintType` defaulted to SHA1 (for example), someone would leave it unspecified in code that creates an SSHFP record, and I think that would be error-prone and confusing — a fingerprint value is nonsense unless its type (and key type) are also known.

There are a few arguments that I think do have useful default values (like the `otherData` field of TSIG) but in my opinion it would be ideal if most of the RR init arguments were mandatory.

Like I said, I would be happy to submit a patch to clean up this part of `dns.py`'s API, but it would change the public interface and have to go through a full deprecation cycle anyway.